From 0ec529e587bc5962e0736d41701dbf24bd5c1651 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Wed, 17 Aug 2022 18:01:01 +0800
Subject: [PATCH 1/3] gpio: s32: force regmap use raw spinlock

The regmap subsystem use mutex lock as its default lock but this will
cause the error that calling sleeping functions in an atomic context
since the irq callbacks "irq_set_type/irq_mask/irq_unmask" for
siul2-s32cc are all using regmap_update_bits.

Add use_raw_spinlock flag to the regmap_config structure siul2_regmap_conf
can switch mutex lock to raw spinlock.

Upstream-Status: Pending

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/gpio/gpio-siul2-s32cc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpio/gpio-siul2-s32cc.c b/drivers/gpio/gpio-siul2-s32cc.c
index 1cef2c90cec3d..5dde1c1975acc 100644
--- a/drivers/gpio/gpio-siul2-s32cc.c
+++ b/drivers/gpio/gpio-siul2-s32cc.c
@@ -536,6 +536,7 @@ static const struct regmap_config siul2_regmap_conf = {
 	.reg_bits = 32,
 	.reg_stride = 4,
 	.cache_type = REGCACHE_FLAT,
+	.use_raw_spinlock = true,
 };
 
 static struct regmap *common_regmap_init(struct platform_device *pdev,
-- 
2.36.1

